src/components/menu/menu-popover.tsx

```diff
@@ -1,14 +1,79 @@
-import { FC, MutableRefObject, useRef } from 'react'
+import { FC, MutableRefObject, useEffect, useRef } from 'react'
```

Comment:
**note:** It's best practice to ensure type-only imports (like FC and MutableRefObject) use the `type` modifier. For example,

```typescript
import { useEffect, useRef } from 'react'
import type { FC, MutableRefObject } from 'react'
// OR...
import { type FC, type MutableRefObject, useEffect, useRef } from 'react' 
```

See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export for more background.